src/module/actor/sheet/popups/casting-item-create-dialog.ts

```diff
@@ -0,0 +1,107 @@
+import { ActorPF2e } from "@actor/index";
+import { SpellPF2e } from "@item";
+import { SpellConsumableItemType } from "@item/consumable/spell-consumables";
+import { OneToTen } from "@module/data";
+import { ErrorPF2e } from "@util";
+
+type FormInputData = {
+    itemTypeOptions?: Object;
+    validLevels?: number[];
+    itemType?: SpellConsumableItemType;
+    level?: OneToTen;
+};
+
+type FormOutputData = {
+    itemType: SpellConsumableItemType;
+    level: OneToTen;
+};
+
+const itemTypeOptions = Object.fromEntries(
+    new Map<SpellConsumableItemType, string>([
+        ["scroll", "PF2E.CastingItemCreateDialog.scroll"],
+        ["wand", "PF2E.CastingItemCreateDialog.wand"],
+        ["cantripDeck5", "PF2E.CastingItemCreateDialog.cantripDeck5"],
+    ])
+);
+
+export class CastingItemCreateDialog extends FormApplication<ActorPF2e> {
+    onSubmitCallback: CastingItemCreateCallback;
+    spell: SpellPF2e;
+    formDataCache: FormOutputData;
+    isCantrip: boolean;
```

Comment:
replacing all internal isCantrip references with this.spell.isCantrip will save you a few lines. There's no real need to cache that.

Reply:
Just thought I'd save lookups but I guess it comes from a set, so it'll be negligible